Aliucord/src/main/java/com/aliucord/coreplugins/badges/BadgesAPI.kt

```diff
@@ -0,0 +1,68 @@
+package com.aliucord.coreplugins.badges
+
+import com.aliucord.*
+import com.aliucord.api.SettingsAPI
+import com.aliucord.settings.delegate
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+internal class BadgesAPI(private val settings: SettingsAPI) {
+    private var SettingsAPI.cacheExpiration by settings.delegate(0L)
+    private var SettingsAPI.cachedBadges by settings.delegate(BadgesInfo(emptyMap(), emptyMap()))
+
+    /**
+     * Get cached badge data or re-fetch from the API if expired.
+     */
+    @OptIn(ExperimentalTime::class)
+    fun getBadges(): BadgesInfo {
+        if (!isCacheExpired()) return settings.cachedBadges
+
+        val data = fetchBadges()
+
+        return if (data != null) {
+            settings.cacheExpiration = System.currentTimeMillis() + Duration.days(1).inWholeMilliseconds
```

Comment:
do we not have the extension vals for duration here? (eg 1.days)

Reply:
I have no idea why, but kotlin deprecated those
it's dumb

Reply:
how recently? because I've used them not that long ago and they weren't deprecated then. But also i don't think it matters for us since we can't really use the newer versions of kotlin anyways